src/commands/hosting-channel-deploy.ts

```diff
@@ -151,7 +131,28 @@ export default new Command("hosting:channel:deploy [channelId]")
 
       logger.info();
       const deploys: { [key: string]: ChannelInfo } = {};
-      sites.forEach((d) => {
+      for (const d of sites) {
+        try {
+          await addAuthDomain(projectId, d.url);
+          logger.debug("[hosting] added auth domain for site", d.site, channelId);
+        } catch (e) {
+          logLabeledWarning(
+            LOG_TAG,
+            marked(
+              `Unable to add channel domain to Firebase Auth. Visit the Firebase Console at ${consoleUrl(
+                projectId,
+                "/authentication/providers"
+              )}`
+            )
+          );
+          logger.debug("[hosting] unable to add auth domain", e);
+        }
+        try {
+          await cleanAuthState(projectId, d.site);
+        } catch (e) {
+          logLabeledWarning(LOG_TAG, "Unable to sync Firebase Auth state.");
+          logger.debug("[hosting] unable to sync auth domain", e);
+        }
```

Comment:
I'd consider moving this into a helper function just to clean up this massive function body we've built...